Kinvey/Kinvey/Kinvey.swift

```diff
@@ -327,9 +327,13 @@ func buildError(
         let debug = json["debug"],
         let description = json["description"]
     {
-        let refreshToken = client.activeUser?.socialIdentity?.kinvey?["refresh_token"] as? String
-        if refreshToken == nil {
-            client.activeUser?.logout()
+        var logoutOnDefer = true
```

Comment:
How is the checking of `refresh_token` related to multi-insert push functionality?

Reply:
we were not handling the `InsufficientCredentials` error before, which should not logout the user.
a scenario detected by the unit tests